syntax = "proto3";

package proto;

service ExchangeService
{
    rpc Ping(EmptyRequest) returns (StringReply) {}

    rpc StreamString(NameRequest) returns (stream StringReply) {}
    rpc StreamInt(NameRequest) returns (stream IntReply) {}

    // rpc BiStreamString(stream EmptyRequest) returns (stream StringReply) {}
    // rpc BiStreamInt(stream EmptyRequest) returns (stream IntReply) {}
}

message EmptyRequest {}

message NameRequest
{
    string name = 1;
}

message StringReply
{
    string msg = 1;
}

message IntReply
{
    int32 msg = 1;
}
